lib/SIL/Linker.cpp

```diff
@@ -197,23 +215,46 @@ bool SILLinkerVisitor::visitFunctionRefInst(FunctionRefInst *FRI) {
   return false;
 }
 
+// Eagerly visiting all used conformances leads to a large blowup
+// in the amount of SIL we read in. For optimization purposes we can defer
+// reading in most conformances until we need them for devirtualization.
+// However, we *must* pull in shared clang-importer-derived conformances
+// we potentially use, since we may not otherwise have a local definition.
+static bool mustDeserializeProtocolConformance(ProtocolConformanceRef c) {
+  return c.isConcrete()
+    && isa<ClangModuleUnit>(c.getConcrete()->getDeclContext()
+                                           ->getModuleScopeContext());
+}
+
 bool SILLinkerVisitor::visitProtocolConformance(
     ProtocolConformanceRef ref, const Optional<SILDeclRef> &Member) {
   // If an abstract protocol conformance was passed in, just return false.
   if (ref.isAbstract())
     return false;
+  
+  bool mustDeserialize = mustDeserializeProtocolConformance(ref);
 
   // Otherwise try and lookup a witness table for C.
   auto C = ref.getConcrete();
-  SILWitnessTable *WT = Mod.lookUpWitnessTable(C);
+  SILWitnessTable *WT = Mod.lookUpWitnessTable(C,
+                                               /*deserialize*/mustDeserialize);
 
   // If we don't find any witness table for the conformance, bail and return
   // false.
   if (!WT) {
     Mod.createWitnessTableDeclaration(
         C, getLinkageForProtocolConformance(
                C->getRootNormalConformance(), NotForDefinition));
-    return false;
+
+    // Adding the declaration may allow us to now deserialize the body.
```

Comment:
may allow us to _not_ deserialize the body?

Reply:
I meant "now", since `lookUpWitnessTable` won't deserialize the witness table body unless a declaration is already in the module.